src/main/kotlin/org/anefdev/mixbuddy/service/MixBuddyService.kt

```diff
@@ -0,0 +1,243 @@
+import se.michaelthelin.spotify.model_objects.specification.PlaylistTrack
+import se.michaelthelin.spotify.model_objects.specification.Track
+import java.net.URI
+import java.util.*
+
+@Service
+@NoArgsConstructor
+class MixBuddyService(
+    private val config: SpotifyConfig,
+    private val playlistParser: PlaylistParser
+) {
+    private val LOGGER: Logger = LoggerFactory.getLogger(MixBuddyService::class.java)
+
+    private var allPlaylists: List<MusicPlaylist>? = null
+    private var playlistParsed: List<MusicTrack>? = null
+    private var currentUser: SpotifyUser? = null
+    private lateinit var spotifyApi: SpotifyApi
+
+    /**
+     * Gets URI for authorization with Spotify.
+     * <br></br>
+     * Sync
+     * @return Authorization code URI
+     */
+    fun getAuthorizationCodeUri(): URI {
+        LOGGER.info("getAuthorisationCodeURI [ Get authorization code URI ... ]")
+        val CALLBACK_URI = SpotifyHttpManager.makeUri(config.callbackUrl)
+
+        this.spotifyApi = SpotifyApi.Builder()
+            .setClientId(config.clientId)
+            .setClientSecret(config.clientSecret)
+            .setRedirectUri(CALLBACK_URI)
+            .build()
+
+        val authorizationCodeUriRequest =
+            spotifyApi.authorizationCodeUri()
+                .scope(
+                    AuthorizationScope.USER_LIBRARY_READ,
+                    AuthorizationScope.USER_READ_PRIVATE,
+                    AuthorizationScope.USER_READ_EMAIL
+                )
+                .show_dialog(true)
+                .build()
+        val uri = authorizationCodeUriRequest.execute()
+        LOGGER.info("getAuthorisationCodeURI [ URI: $uri ]")
+        LOGGER.info("getAuthorisationCodeURI [ OK ]")
+        return uri
+    }
+
+    /**
+     * Gets and saves the access and refresh tokens in SpotifyApi object.
+     * <br></br>
+     * Sync
+     * @param code Authorization code for Spotify
+     */
+    @SneakyThrows
+    fun setAuthorizationToken(code: String) {
+        LOGGER.info("setAuthorizationToken [ Get authorization token ...")
+        LOGGER.info("setAuthorizationToken [ Authorization code: $code ]")
+
+        val authorizationCodeRequest = spotifyApi.authorizationCode(code).build()
+        val authorizationCodeCredentials = authorizationCodeRequest.execute()
+        // Set access and refresh token for further "spotifyApi" object usage
+        spotifyApi.accessToken = authorizationCodeCredentials.accessToken
+        spotifyApi.refreshToken = authorizationCodeCredentials.refreshToken
+
+        LOGGER.info("setAuthorizationToken [ Access token: " + spotifyApi.accessToken + " ]")
+        LOGGER.info("setAuthorizationToken [ Refresh token: " + spotifyApi.refreshToken + " ]")
+    }
+
+    /**
+     * Loads all playlists of the current user.
+     * <br></br>
+     * Sync
+     * @return List of user's playlists
+     */
+    @SneakyThrows
+    fun loadAllUsersPlaylists(): List<MusicPlaylist>? {
```

Comment:
this doesnt seem like a good approach, in what cases list can be `null`?
i see that it can be empty but not null (from a logic standpoint)